src/lib/ui/thermostat/ClimatePreset.tsx

```diff
@@ -0,0 +1,299 @@
+        type: 'validate',
+        message: 'Preset with this key already exists.',
+      })
+      return
+    }
+
+    const name = state.name.replace(/\s+/g, ' ').trim()
+
+    const body = {
+      climate_preset_key: state.key,
+      device_id: device.device_id,
+      name: name === '' ? undefined : name,
+      cooling_set_point_fahrenheit: state.coolPoint,
+      heating_set_point_fahrenheit: state.heatPoint,
+      fan_mode_setting: state.fanMode,
+      cooling_set_point_celsius:
+        typeof state.coolPoint === 'number'
+          ? toCelcius(state.coolPoint)
+          : undefined,
+      heating_set_point_celsius:
+        typeof state.heatPoint === 'number'
+          ? toCelcius(state.heatPoint)
+          : undefined,
+      hvac_mode_setting: state.hvacMode,
+    }
+
+    if (isCreate) {
+      createMutation.mutate(body)
+    } else {
+      updateMutation.mutate({
+        ...body,
+        manual_override_allowed: true,
+      })
+    }
+  }
+
+  return (
+    <div
+      {...attrs}
+      className={classNames('seam-thermostat-climate-preset', attrs.className)}
+    >
+      <ContentHeader title={preset.display_name} onBack={onBack} />
+
+      <div className='seam-main'>
+        <form
+          onSubmit={(e) => {
+            void handleSubmit(onSubmit)(e)
+          }}
+        >
+          {isCreate && (
+            <FormField>
+              <InputLabel>Key</InputLabel>
+              <TextField
+                size='large'
+                clearable
+                hasError={errors.key != null}
+                helperText={errors.key?.message}
+                inputProps={{
+                  ...register('key', {
+                    required: 'required',
+                    maxLength: {
+                      value: 20,
+                      message: 'max 20 chars',
+                    },
+                    minLength: {
+                      value: 3,
+                      message: 'min 3 chars',
+                    },
+                    validate(value) {
+                      const fixedValue = value.replace(/\s+/g, '').trim()
+                      return !otherPresets.some(
+                        (other) => other.climate_preset_key === fixedValue
+                      )
+                    },
+                  }),
+                }}
+              />
+            </FormField>
+          )}
+
+          <FormField>
+            <InputLabel>Display Name (Optional)</InputLabel>
+            <TextField
+              size='large'
+              clearable
+              hasError={errors.name != null}
+              helperText={errors.name?.message}
+              inputProps={{
+                ...register('name', {
+                  maxLength: {
+                    value: 20,
+                    message: 'max 20 chars',
+                  },
+                  minLength: {
+                    value: 3,
+                    message: 'min 3 chars',
+                  },
```

Comment:
Move to `t`

There are other things that need to be moved to `t`, I'll stop commenting on each one.